### ISSUE TITLE
Packaged application gives error message: Could not resolve a binding for http://schemas.xmlsoap.org/wsdl/soap/

### ISSUE THREAD

**Issue by matheusdallrosa** [quarkiverse/quarkus-cxf#194]
When using the application with dev mode ( without packaging ) everything works perfectly. But after we package it as an uber-jar it gives the following error:
````
Could not resolve a binding for http://schemas.xmlsoap.org/wsdl/soap/
````

We are using quarkus-cxf version 0.5 and quarkus version 1.11.3.Final.

If you need more information, please let me know.

**Comment by matheusdallrosa** [comment]
Using the default packaging type for the version 1.11.3.Final it works properly.

**Comment by matheusdallrosa** [comment]
But it is very important to me that it also works when packaging the application as an uber-jar.

**Comment by dufoli** [comment]
hmmm
I think we need to add 
META-INF/cxf/cxf-servlet.xml
META-INF/cxf/cxf-extension-jaxws.xml
to NativeImageResourceBuildItem in /QuarkusCxfProcessor

@matheusdallrosa  can you try with this patch on master ?

**Comment by dufoli** [comment]
maybe because SoapBindingFactory and XMLBindingFactory must be loaded
so it mean that cxf-rt-bindings-soap must be included in dependencies, is it done ?

**Comment by dufoli** [comment]
another lead : https://cxf.apache.org/docs/bundling-cxf-into-single-jar-with-maven-shade-plugin.html

**Comment by matheusdallrosa** [comment]
> hmmm
> I think we need to add
> META-INF/cxf/cxf-servlet.xml
> META-INF/cxf/cxf-extension-jaxws.xml
> to NativeImageResourceBuildItem in /QuarkusCxfProcessor
> 
> @matheusdallrosa can you try with this patch on master ?
> 
> I just found this info on stack overflow so maybe wrong.
> Anyway, I see that error come from : https://github.com/apache/cxf/blob/75fb6bb56d82f72771a9ee6ecab5d36168303f51/rt/frontend/simple/src/main/java/org/apache/cxf/frontend/AbstractWSDLBasedEndpointFactory.java#L324

The exception definitely comes from the method you mentioned.
I will try to add the files you mentioned.

**Comment by shumonsharif** [comment]
@matheusdallrosa @dufoli This is a duplicate of https://github.com/quarkiverse/quarkus-cxf/issues/82 ... until https://github.com/quarkusio/quarkus/issues/5677 is fixed, the only good solution would be to create your own pre-shaded CXF bundle, though that may be a bit complex.

**Comment by matheusdallrosa** [comment]
> maybe because SoapBindingFactory and XMLBindingFactory must be loaded
> so it mean that cxf-rt-bindings-soap must be included in dependencies, is it done ?

Tried it and it did not work.

**Comment by shumonsharif** [comment]
Also, you may have looked into it already - it may be worth exploring the [fast-jar option](https://quarkus.io/guides/maven-tooling#fast-jar), which is preferred, instead of uber-jar.  Just keep in mind the executable jar will end up in the target/quarkus-app folder.  It works fine (without the issues above) - and provides slightly better performance.

**Comment by matheusdallrosa** [comment]
> Also, you may have looked into it already - it may be worth exploring the [fast-jar option](https://quarkus.io/guides/maven-tooling#fast-jar), which is preferred, instead of uber-jar. Just keep in mind the executable jar will end up in the target/quarkus-app folder. It works fine (without the issues above) - and provides slightly better performance.

Yes, i would like to move to fast-jar, but at the moment is not possible for client's reasons.

**Comment by matheusdallrosa** [comment]
So, just to put everything together, from what I understood from #82 and [https://github.com/quarkusio/quarkus/issues/5677](https://github.com/quarkusio/quarkus/issues/5677) is that there is some dependency creating 
a set of files inside META-INF folder before the correct set of files are created inside the META-INF folder, and because they have the same name the ones written first have priority, and because of that we don't have the correct files. Is my understanding correct?

**Comment by shumonsharif** [comment]
Yes, your thinking is along the right lines ... an uber-jar merges everything into a single jar; in case of a path collision, only one file currently gets picked.  Instead of only one file getting picked when there is a conflict, all the conflicting files need to be merged using transformers (this includes property and xml files).  You can take a look at the following link in case that's a route you want to look at:  https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html

**Comment by matheusdallrosa** [comment]
How can we know which files we need to "give priority"?
Maybe 

````
META-INF/cxf/cxf-servlet.xml
META-INF/cxf/cxf-extension-jaxws.xml
````

Would be enough?

**Comment by matheusdallrosa** [comment]
We have the following file conflicts while packaging the app:
````
[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-frontend-simple::jar:3.4.2(compile) will be ignored. 
Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-ws-policy::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-ws-addr::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-frontend-jaxws::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-transports-http::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-wsdl::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-databinding-jaxb::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-bindings-soap::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-bindings-xml::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/wsdl.plugin.xml entry from org.apache.cxf:cxf-rt-bindings-xml::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-rt-bindings-soap::jar:3.4.2(compile)

[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry org/apache/cxf/endpoint/dynamic/simple-binding.xjb entry from org.apache.cxf:cxf-rt-frontend-simple::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-rt-databinding-jaxb::jar:3.4.2(compile)
````

Any advice on strategies to solve them?

**Comment by shumonsharif** [comment]
Hey @matheusdallrosa I think we've already provided several links that will be helpful for you.  I would suggest you start with something like this:  https://cxf.apache.org/docs/bundling-cxf-into-single-jar-with-maven-shade-plugin.html  ... beyond this, I'm afraid I don't have any generic advice, but let us know if you have any specific questions, and please do keep us posted if you can find a solution.

**Comment by dufoli** [comment]
hmm maybe there is a solution by using GeneratedResourceBuildItem with List<UberJarRequiredBuildItem> and if list is not empty we merge all cxf-extension xml file together. I think it will help a lot. no?

**Comment by dufoli** [comment]
btw wsdl.plugin.xml must be merged too:
https://github.com/apache/cxf/blob/master/rt/bindings/xml/src/main/resources/META-INF/wsdl.plugin.xml
https://github.com/apache/cxf/blob/master/rt/bindings/soap/src/main/resources/META-INF/wsdl.plugin.xml

**Comment by dufoli** [comment]
@shumonsharif I have no time to test that but if you have on your own... it is a lead to have equivalent of plugin to merge resources. After all, processor is a precompilation so merge file must be doable with it.

**Comment by shumonsharif** [comment]
Fair points @dufoli - definitely worth exploring the approach you suggested, though I'm not sure what conflicts may arise with the existing Quarkus uber jar packaging mechanisms.  Will take a look and keep you posted!

**Comment by dufoli** [comment]
I have implement the idea, If you can test the branch on your side before merge...

**Comment by shumonsharif** [comment]
@dufoli Testing this now, I will keep you posted with findings.  Keep in mind though, I'm not sure why exactly - but I never encountered this issue with native builds - native has always worked fine for me.  I have seen the issue with uber jars only.

**Comment by shumonsharif** [comment]
@dufoli The issue remains; getting the error below using the uberjar branch.

```
17:32:09 ERROR [io.qu.ru.Application] (main) Failed to start application (with profile prod): org.apache.cxf.BusException: No binding factory for namespace http://www.w3.org/2003/05/soap/bindings/HTTP/ registered.
        at org.apache.cxf.bus.managers.BindingFactoryManagerImpl.getBindingFactory(BindingFactoryManagerImpl.java:93)
        at org.apache.cxf.frontend.AbstractWSDLBasedEndpointFactory.createBindingInfo(AbstractWSDLBasedEndpointFactory.java:348)
        at org.apache.cxf.jaxws.JaxWsServerFactoryBean.createBindingInfo(JaxWsServerFactoryBean.java:182)
        at org.apache.cxf.frontend.AbstractWSDLBasedEndpointFactory.createEndpointInfo(AbstractWSDLBasedEndpointFactory.java:268)
        at org.apache.cxf.frontend.AbstractWSDLBasedEndpointFactory.createEndpoint(AbstractWSDLBasedEndpointFactory.java:146)
        at org.apache.cxf.frontend.ServerFactoryBean.create(ServerFactoryBean.java:168)
        at org.apache.cxf.jaxws.JaxWsServerFactoryBean.create(JaxWsServerFactoryBean.java:211)
        at io.quarkiverse.cxf.transport.CxfHandler.<init>(CxfHandler.java:141)
        at io.quarkiverse.cxf.CXFRecorder.initServer(CXFRecorder.java:184)
        at io.quarkus.deployment.steps.QuarkusCxfProcessor$startRoute-273993731.deploy_0(QuarkusCxfProcessor$startRoute-273993731.zig:263)
        at io.quarkus.deployment.steps.QuarkusCxfProcessor$startRoute-273993731.deploy(QuarkusCxfProcessor$startRoute-273993731.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:872)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:100)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
```

**Comment by dufoli** [comment]
@famod @gastaldi @gsmet do you know a way to transform resource to merge ressource for Uber jar and native to avoid to have the duplicate entry issue.

I do not found anything in doc to merge resource whereas it seems doable in quarkus build. I have try to use generatedRessourceBuildItem but do not worked. You can see code in pull request attached to this issue. 


[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/cxf/bus-extensions.txt entry from org.apache.cxf:cxf-rt-databinding-jaxb::jar:3.4.2(compile) will be ignored. Existing file was provided by org.apache.cxf:cxf-core::jar:3.4.2(compile)

**Comment by gastaldi** [comment]
Looks like this file is already inside the `org.apache.cxf:cxf-core::jar:3.4.2` JAR. Why are you regenerating it?

**Comment by dufoli** [comment]
@gastaldi indeed this /META-INF/cxf/bus-extension.txt resource is present in all jar of cxf (core but not only).
So we end with duplicate resource.

it is a basic text file and we can have single file which contain all line merged from different files.
I know that mave shade plugin can do the trick. But I think that we can do it with quarkus, it is a pre compilation afterall so combine resource to a new one must be doable. That s why I think to generated resource build item. The issue is that I do not find how delete old resources to avoid duplicate resource errors and seems not working as expected.

here is the process in cxf.
https://cxf.apache.org/docs/bundling-cxf-into-single-jar-with-maven-shade-plugin.html

here is the full list:
 core/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/wsdl/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/ws/rm/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/rs/sse/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/ws/mex/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/ws/addr/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/ws/policy/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/management/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/ws/security/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/bindings/xml/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/bindings/soap/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/bindings/corba/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/frontend/jaxrs/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/frontend/jaxws/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/rs/description/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/jms/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/udp/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/frontend/simple/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/http/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/databinding/jaxb/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/features/logging/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/features/metrics/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/local/src/main/resources/META-INF/cxf/bus-extensions.txt
 systests/ws-security/src/test/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/http-hc/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/websocket/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/http-jetty/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/javascript/javascript-rt/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/http-undertow/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/http-netty/netty-client/src/main/resources/META-INF/cxf/bus-extensions.txt
 rt/transports/http-netty/netty-server/src/main/resources/META-INF/cxf/bus-extensions.txt
 services/ws-discovery/ws-discovery-service/src/main/resources/META-INF/cxf/bus-extensions.txt

**Comment by famod** [comment]
@geoand may I kindly ask you to have a look at ⬆️? Do you know a trick that might help here? Thanks!
It seems https://github.com/quarkusio/quarkus/issues/5677 would help here but I guess it's not a low hanging fruit.

**Comment by gastaldi** [comment]
I think using the shade plugin could be a good strategy here. If the extension depends on the original JARs the duplicated file will always be present

**Comment by geoand** [comment]
> @geoand may I kindly ask you to have a look at ? Do you know a trick that might help here? Thanks!
> It seems [quarkusio/quarkus#5677](https://github.com/quarkusio/quarkus/issues/5677) would help here but I guess it's not a low hanging fruit.

Given that this only occurs with uber-jar, I seriously doubt I will ever get around to it :). Others are welcome to try of course and I can try and help

**Comment by dufoli** [comment]
we have same issue with native too. That s why I was thinking to have a feature in quarkus for resource merging.

**Comment by gastaldi** [comment]
@dufoli perhaps this can be solved by introducing a `UberJarResourceMergeBuildItem` like the following:

```java
/**
 * Merge a resource from multiple JARs when building an Uber Jar
 */
public final class UberJarResourceMergeBuildItem extends MultiBuildItem {

    private final String path;

    public UberJarResourceMergeBuildItem(String path) {
        this.path = path;
    }

    public String getPath() {
        return path;
    }
}
```

And then we change `JarResultBuildStep` to compute these while building the UberJar. Shouldn't be too hard to implement I think.

**Comment by gastaldi** [comment]
@dufoli I created a PR in https://github.com/quarkusio/quarkus/pull/17077 introducing `UberJarResourceMergedBuildItem`. Can you give it a try if that works for you?

**Comment by shumonsharif** [comment]
@dufoli @gastaldi As mentioned previously, the issue does not appear to occur in native mode.  I'm not sure if I'm missing something, but my testing has repeatedly shown it to be limited to uber jar mode only.  Fast jar mode is also not an issue, even though we're not merging any resources there.

It's a really simple test to see what's getting included in the native image, and the output for the snippet shows all the bus-extensions.txt files getting bundled in the native image.

```java
import io.quarkus.runtime.StartupEvent;
import org.jboss.logging.Logger;

import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.event.Observes;
import java.io.IOException;
import java.net.URL;
import java.util.Collections;
import java.util.Enumeration;

@ApplicationScoped
public class CXFResourceLogger {

    private static final Logger LOGGER = Logger.getLogger(CXFResourceLogger.class);

    void onStart(@Observes StartupEvent ev) throws IOException {

        LOGGER.info("Identify resources in native image ...");

        ClassLoader loader = Thread.currentThread().getContextClassLoader();
        Enumeration<URL> urls = loader.getResources("META-INF/cxf/bus-extensions.txt");

        Collections.list(urls).forEach((url) ->
            LOGGER.info(url)
        );
    }

}
```

Anyway, just wanted to make note of this in case it helps in any way, since we seem to be having some disagreement on the native mode.

Finally - I'm happy to help with this issue @dufoli so please keep me posted if there's anything I can do!

**Comment by famod** [comment]
> @dufoli I created a PR in [quarkusio/quarkus#17077](https://github.com/quarkusio/quarkus/pull/17077) introducing `UberJarMergedResourceBuildItem`. Can you give it a try if that works for you?

FTR: This is included in Quarkus 1.13.4.Final which has been released today.

**Comment by gastaldi** [comment]
@shumonsharif @dufoli take a look at https://github.com/quarkusio/quarkus/blob/main/core/test-extension/deployment/src/main/java/io/quarkus/extest/deployment/UberJarConfigBuildStep.java#L14

That should be enough to have the files merged in the Uber Jar

**Comment by shumonsharif** [comment]
@gastaldi Thank you; I think your PR should partially solve the issue we're having.  I noticed you had this ...
https://github.com/gastaldi/quarkus/blob/3f079855681195af7bf96d149ee6c779a4f16d88/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java#L1172

... so we should be half way there.  :-)  The XML merge will also need to be included at some point for this issue to be addressed in full; is that something you may be able to help with?  I can look into it, though I'll need get set up with the quarkus code base first!

**Comment by gastaldi** [comment]
@shumonsharif sure, I'll send a PR soon and ping you in the PR for testing, if you like

**Comment by shumonsharif** [comment]
@gastaldi absolutely, kindly keep us posted.

**Comment by dufoli** [comment]
thanks @shumonsharif  Iwas thinking issue happend in native mode because of some stack trace but indeed it was the providerImpl wich was missing in resource. So, yes you are right it is only in uber jar.
For uberjar, merging of file is great. But if we can have other kind of merging as xml. If we can have something where we transform.
I was thinking to something like defining
TransformResourceBuildItem(String name, (resources) => {return Byte[]});
or 
TransformResourceBuildItem(filterFunction, String name, (resources) => {return Byte[]});

old resources are deleted and new one is produce with byte as content .

**Comment by gastaldi** [comment]
I am not sure we can define functions inside BuildItems. We could perhaps have the resulting contents of the merged resources at least. 

I have created https://github.com/quarkusio/quarkus/pull/17196 for combining XML files properly, but perhaps we could have a build item that replaces the entire resource with something else.

**Comment by gastaldi** [comment]
> For uberjar, merging of file is great. But if we can have other kind of merging as xml, json,.... If we can have something where we transform.
> I was thinking to something like defining
> TransformResourceBuildItem(String name, (resources) => {return Byte[]});
> or
> TransformResourceBuildItem(filterFunction, String name, (resources) => {return Byte[]});
> 
> old resources are deleted and new one is produce with byte as content .
> 
> And even, if we do it for uberJar, it can be for other thing. Maybe optimize resource with mignification or whatever.... so maybe name it with more generic name.

@dufoli I think you described `GeneratedResourceBuildItem`?

**Comment by gastaldi** [comment]
Okay, now we're getting somewhere. Producing `GeneratedResourceBuildItem`s should be what you' re looking for: https://github.com/quarkusio/quarkus/pull/17199

**Comment by shumonsharif** [comment]
@gastaldi This is great, about to start testing your new PR, will keep you posted, thanks again.

**Comment by shumonsharif** [comment]
@gastaldi Testing has been successful, thanks for your help with this.  We're finally in a good spot!  :)

@dufoli I've attached a patch of the changes I made for testing purposes, feel free to review and integrate ... we'll need to wait for the next Quarkus release before we can do much with this anyway.  Also attached the merged file contents for `META-INF/cxf/bus-extensions.txt` and `META-INF/wsdl.plugin.xml`, extracted from the uber jar.

[quarkus-cxf-194.patch.txt](https://github.com/quarkiverse/quarkus-cxf/files/6469965/quarkus-cxf-194.patch.txt)

[merged-bus-extensions.txt](https://github.com/quarkiverse/quarkus-cxf/files/6469957/merged-bus-extensions.txt)
[merged-wsdl.plugin.xml.txt](https://github.com/quarkiverse/quarkus-cxf/files/6469963/merged-wsdl.plugin.xml.txt)

**Comment by dufoli** [comment]
@shumonsharif  I check the patch it is great but I do not like to add new dependency just to combine xml so I just update the uberjar branch with xml combine with standard xml reader/writer. based on your work. I am not a big fan of current version either because it add a lot of code for not so much.... I will not merged right away to discuss about it.

**Comment by shumonsharif** [comment]
Hey @dufoli these are the commands I ran after running `mvn clean install -Dquarkus.package.type=uber-jar`

`unzip -p quarkus-cxf-integration-tests-0.8.0-SNAPSHOT-runner.jar 'META-INF/cxf/bus-extensions.txt'`
`unzip -p quarkus-cxf-integration-tests-0.8.0-SNAPSHOT-runner.jar 'META-INF/wsdl.plugin.xml'`

**Comment by dufoli** [comment]
I have just test and the uberjar work too ;-) @shumonsharif

**Comment by dufoli** [comment]
I have tested with 2.0.0Alpha3 and it is working perfectly!

**Comment by dufoli** [comment]
if we switch to 2.0 branch of quarkus we have to wait at least 1 month the final release
current release alpha 2
we can expect at least 2 RC before final so it mean a month and a half. So I will not switch to 2.0 release and hope next 1.3 final release will get the back port of this commit.

**Comment by shumonsharif** [comment]
@dufoli It looks like the issue was merged for 1.13.5.Final, so the back port is there (Honor GenerateResourceBuildItem while building UberJar)
https://github.com/quarkusio/quarkus/milestone/126?closed=1

I think 1.13.5.Final will be released very soon, so glad to see your merged this in!

**Comment by gastaldi** [comment]
@shumonsharif yes, 1.13.5.Final should be hopefully out on May 26th
